### PR TITLE
Remove DDP teardown move model to CPU

### LINKED ISSUE #10294: teardown function casts model to cpu after testing
https://github.com/PyTorchLightning/pytorch-lightning/blob/6e124e7207f6459cb43f540cfb5a1c6cc9b00f7a/pytorch_lightning/plugins/training_type/single_device.py#L83-L88

Issue: This part of code forces model to be on CPU after the testing is finished

expected behaviour: the teardown logic after testing should not change the model device

reproducible code snippet:

```
    # model.device --> 'cuda'
    test_loader = data_module.test_dataloader()
    tester = pl.Trainer(precision=precision, gpus=gpus, logger=False)
    test_results = tester.test(model, test_loader)
    # model.device --> 'cpu'
```

cc @borda @akihironitta

### PR DESCRIPTION
**Pull request by four4fish**
## What does this PR do?

I didn't see logic depends on ttp teardown to move model to cpu after fit or test

Other than the error mentioned in the issue, we saw internal failures for FSDP. Model is too big doesn't fit into cpu which caused OOM. 

Fixes #10294 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by justusschock** [comment]
@four4fish The idea behind this is that the model is usually on CPU before fit and thus after the entrypoint we want to reset everything back to it's original state (except the weights of course ^^) to have a minimal impact.

**Comment by awaelchli** [comment]
To address the linked issue, wouldn't we need to move the model back to the device it was originally on?

**Comment by four4fish** [comment]
> To address the linked issue, wouldn't we need to move the model back to the device it was originally on?

Hi @awaelchli, I thought user is `Issue: This part of code forces model to be on CPU after the testing is finished; expected behaviour: the teardown logic after testing should not change the model device` which means we don't want to move the model to cpu?

Also we saw failures in internal FB for FSDP. Model is too big doesn't fit into cpu which caused OOM.
For DDP, all rank trying to move model back to cpu which cause OOM too (I need go verify this more)

**Comment by awaelchli** [comment]
Given the code snippet in the linked issue and the comment from @ananthsub https://github.com/PyTorchLightning/pytorch-lightning/issues/10294#issuecomment-956455783 I interpreted that the goal was to leave the model on the device it was originally on (before fit). This is strictly speaking not the same as not moving it back to CPU.

**Just one important remark everyone should be aware of:** 
The code in the teardown method moving the model to CPU is there for a reason, I am 99% sure. The problem is that it is not documented, so we don't remember why. There is a chance that if we make the change then we also break what this was fixing originally. There is also a chance that it has become redundant. 
In any case, it's a breaking change: The user needs to add a line `model.cpu()` to preserve the behavior if they need.

> Also we saw failures in internal FB for FSDP. Model is too big doesn't fit into cpu which caused OOM.
> For DDP, all rank trying to move model back to cpu which cause OOM too (I need go verify this more

Have we considered changing `teardown()` for FSDP?
